src/dtproject/types.py

```diff
@@ -157,6 +157,15 @@ class DevContainerConfiguration(dict):
     service: Optional[str] = None
     workspaceFolder: Optional[str] = None
 
+@dataclasses.dataclass
+class Hook():
+    command: str = ""
+    required: Optional[bool] = False
+
+@dataclasses.dataclass
+class LayerHooks(DataClassLayer):
+    version: int
+    hooks: DefaultDict[str, List[Hook]]
```

Comment:
This needs to be initialized to a `dataclasses.field(default_factory=lambda: defaultdict(list))` so that querying a hook type (e.g., `pre-build`) on a project that does not define hooks will return an empty list of `Hooks`.

Reply:
Testing it in a Python 3.12.0 venv the output of `dataclasses.asdict(LayerHooks)` is `{'hooks': defaultdict(<class 'list'>, {})}` rather than throwing the `TypeError: first argument must be callable or None` error.